File: maths/codes/download/Prime nos bw two nos.cpp
#include <iostream>

int basic_prime[25] = {3, 5, 7, 11, 13, 15, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97};

int get_integer(string msg = "Number: ") {
    int i;
    cout << msg;
    cin >> i;
    return i;
}

bool isodd(int num) {
    bool odd = false;
    if (num % 2) {
        odd = true;
    }
    return odd;
}

bool iseven(int num) {
    bool even = true;
    if (num % 2) {
        even = false;
    }
    return even;
}

bool isprime(int num) {
    bool prime = true;
    if (iseven(num) and num / 2 != 1) {
        prime = false;
    }
    if (isodd(num)) {
        for (int i = 0; i < 25; i++) {
            if (num % basic_prime[i] == 0 and num / basic_prime[i] != 1) {
                prime = false;
            }
        }
        if (prime) {
            for (int i = 101; i < (sqrt(num) + 1); i++) {
                if (num % i == 0) {
                    prime = false;
                    break;
                }
            }
        }
    }
    return prime;
}


using namespace std;

int main() {
    while (true) {
        int ini_no = get_integer("Initial no: ");
        int final_no = get_integer("Final no: ");

        int total = 0;
        cout << "Prime no.s in range from " << ini_no << " to " << final_no << " are: \n";
        for (int i = ini_no; i <= final_no; i++) {
            if (isprime(i) and i != 1) {
                cout << i << endl;
                total++;
            }
        }
        if (total == 0) {
            cout << "None\n";
        }
        else {
            cout << "Total = " << total << endl;
        }
        cout << endl;
    }
}

File: maths/codes/download/Prime nos bw two nos.py
def get_integer(msg="Number: ", wrng_msg="Invalid Input"):
    try:
        i = int(input(msg))
    except ValueError:
        print(wrng_msg)
        return get_integer(msg, wrng_msg)
    else:
        return i

def isodd(num):
    odd = True
    if num % 2 == 0:
        odd = False
    return odd


def iseven(num):
    even = True
    if num % 2 != 0:
        even = False
    return even


def isprime(num):
    prime = True
    if iseven(num) and num / 2 != 1:
        prime = False
    if isodd(num):
        for i in basic_prime:
            if num % i == 0 and num / i == 1:
                break
            elif num % i == 0 and num / i != 1:
                prime = False
        if prime:
            for i in range(101, int(math.sqrt(num) + 1), 2):
                if num % i == 0:
                    prime = False
                    break
    return prime


from time import time

while True:
    ini_no = get_integer('Initial no: ')
    final_no = get_integer('Final no: ')
    
    total = 0
    print(f'Prime no.s in range from {ini_no} to {final_no} are: ')
    a = time()
    for i in range(ini_no, final_no + 1):
        if isprime(i) and i != 1:
            #print(i)
            total += 1
    if total == 0:
        print('None')
    else:
        print(f"Total = {total}")
    print(f"Time = {time() - a}\n")
        
    print('')
    if input("Wanna quit?(y/n): ") == 'y':
        break
